apps/account-api/src/services/delegation.service.ts

```diff
@@ -47,36 +46,31 @@ export class DelegationService {
     throw new Error('Invalid msaId.');
   }
 
-  async getRevokeDelegationPayload(accountId: string, providerId: string): Promise<RevokeDelegationPayloadResponseDto> {
-    if (!isAddress(accountId)) {
-      throw new HttpException('Invalid accountId', HttpStatus.BAD_REQUEST);
-    }
+  async getRevokeDelegationPayload(
+    accountId: string,
+    providerMsaId: string,
+  ): Promise<RevokeDelegationPayloadResponseDto> {
     const msaId = await this.blockchainService.publicKeyToMsaId(accountId);
     if (!msaId) {
       throw new HttpException('MSA ID for account not found', HttpStatus.NOT_FOUND);
     }
 
-    if (providerId) {
-      const isValidProviderId = await this.blockchainService.isValidMsaId(providerId);
-      if (!isValidProviderId) {
-        throw new HttpException('Invalid provider', HttpStatus.BAD_REQUEST);
-      }
-
-      const providerInfo = await this.blockchainService.getProviderToRegistryEntry(providerId);
+    if (providerMsaId) {
+      const providerInfo = await this.blockchainService.getProviderToRegistryEntry(providerMsaId);
```

Comment:
Are we sure that this is the change we want? Previously, we threw 2 different errors:
* `BAD_REQUEST('Invalid provider')` if the ID supplied for the Provider is not a valid MSA
* `BAD_REQUEST('Supplied ID not a Provider')` if the ID supplied IS an MSA, but IS NOT a registered Provider

I think having 2 different error strings returned is useful to a client in understanding _why_ their request failed.

Reply:
I might be missing something but I think we only got rid of the first one not the second

Reply:
Right; we've consolidated 2 errors -> 1 error. Do we want to do that?